packages/cli/src/lib/generators/CodeGenerator.ts

```diff
@@ -0,0 +1,104 @@
+import { BuildConfig } from "../Compiler";
+import { SchemaComposer } from "../SchemaComposer";
+
+import chalk from "chalk";
+import fs, { readFileSync } from "fs";
+import path from "path";
+import Mustache from "mustache";
+import {
+  OutputDirectory,
+  OutputEntry,
+  writeDirectory,
+} from "@web3api/schema-bind";
+import { TypeInfo, parseSchema } from "@web3api/schema-parse";
+
+export class CodeGenerator {
+  private _schema: string | undefined = "";
+  constructor(private _templateFile: string, private _config: BuildConfig) {}
```

Comment:
Let's use a `CodeGeneratorConfig` here that only has the specific properties required for this class' methods.

Reply:
With this we can make "templateFile" apart of the config.

Reply:
Also don't forget to rename "templateFile" to "GenerationFile" or maybe "GenerationRunner", depending on what you choose.

Reply:
GenerationScript?

Reply:
GenerationFile looks good. Sth like `GeneratorScript` would be more suitable for `.gen.js` files, but it wouldn't make sense for `.gen.yaml` files.